File: contests/abc106/abc106_b.py
# B - 105
# https://atcoder.jp/contests/abc106/tasks/abc106_b

n = int(input())

count = 0
odd_count = 0
for i in range(1, n+1):
    if i % 2 != 0:
        for j in range(1, i+1):
            if i % j  == 0:
                odd_count += 1
        if odd_count == 8:
            count += 1
        odd_count = 0

print(count)

File: contests/abc106/abc106_b.cpp
// B - 105
// https://atcoder.jp/contests/abc106/tasks/abc106_b

#include <iostream>
using namespace std;

int main(){
    int n; cin >> n;
    int ans=0;
  
    for(int i=1;i<=n;i+=2){
        int count = 0; 
        for(int j=1;j<=i;j++)
            if(i%j==0) count++;
        if(count==8) ans++;
    }  
    cout << ans << endl;
}
